chase/frontend/components/speakers_list/add_speaker.tsx

```diff
@@ -0,0 +1,168 @@
+import {
+  AutoComplete,
+  AutoCompleteCompleteEvent,
+} from "primereact/autocomplete";
+import getCountryNameByCode from "@/misc/get_country_name_by_code";
+import { useI18nContext } from "@/i18n/i18n-react";
+import { Button } from "primereact/button";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { faPlus, faTimes } from "@fortawesome/free-solid-svg-icons";
+import { SmallFlag } from "../flag_templates";
+import { CountryCode } from "@/custom_types";
+
+interface CountryData {
+  alpha3: CountryCode;
+  name: string;
+}
+
+/**
+ * This component is used to display the overlay to add a speaker to the Speakers List on the Speakers List Page for chairs.
+ * It uses the AutoComplete Component from PrimeReact.
+ * The overlay is displayed when the user clicks on the Add Speaker Button.
+ * The user can search for a country by name or by code.
+ * The user can select a country from the list of suggestions.
+ * The user can click on the Add Speaker Button to add the speaker and keep the overlay open, or click on the Add and Close Button to add the speaker and close the overlay.
+ * Instead of clicking on the Add Speaker Button, the user can press the Enter key after selecting a country to trigger the same action.
+ * Note: Not only countries can be added to the Speakers List, but also Non-State Actors and als UN Staff like the Secretary-General (Country-Code: unm / unw (male/female))
+ */
+
+// TODO add toast functionality for when a speaker is added or already on the list
+// TODO add warning when a speaker is added to the list and the list is closed
+
+export default function AddSpeakerOverlay({
+  listOfAllCountries,
+  listClosed,
+  closeOverlay,
+}: {
+  listOfAllCountries: CountryCode[];
+  listClosed: boolean;
+  closeOverlay: () => void;
+}) {
+  const { LL, locale } = useI18nContext();
+
+  const [countries, setCountries] = useState<CountryData[] | null>(null);
+  const [query, setQuery] = useState<string>("");
+  const [filteredCountries, setFilteredCountries] = useState<CountryData[]>([]);
+  const [selectedCountry, setSelectedCountry] = useState<CountryData | null>(
+    null,
+  );
+
+  useEffect(() => {
+    if (!listOfAllCountries) return;
+    const countryData: CountryData[] = listOfAllCountries.map(
+      (country: CountryCode) => {
+        return {
+          alpha3: country,
+          name: getCountryNameByCode(country, locale),
+        };
+      },
+    );
+    setCountries(countryData);
+  }, [listOfAllCountries, locale]);
+
+  useEffect(() => {
+    if (selectedCountry) {
+      setQuery(selectedCountry.name);
+    }
+  }, [selectedCountry]);
+
+  useEffect(() => {
+    if (query === "") {
+      setSelectedCountry(null);
+    } else {
+      const country = countries?.find((country) => country.name === query);
+      if (country) {
+        setSelectedCountry(country);
+      }
+    }
+  }, [query]);
+
+  const searchCountry = (event: AutoCompleteCompleteEvent) => {
```

Comment:
We could think about implementing fuzzy search for this. Have a look at https://fusejs.io/

Reply:
Good Idea. I've already worked with Fuzzy Search in Python, I will look into that.

Reply:
Put this into its own issue, resolving